src/pokemon.c

```diff
@@ -4,58 +4,322 @@
+				    bool *seguir_leyendo)
+{
+	(info_pokemon->cantidad_pokemones)++;
+	pokemon_t **puntero_auxiliar =
+		realloc(info_pokemon->pokemones,
+			(long unsigned int)(info_pokemon->cantidad_pokemones) *
+				sizeof(pokemon_t));
+	if (puntero_auxiliar != NULL) {
+		info_pokemon->pokemones = puntero_auxiliar;
+		info_pokemon->pokemones[(info_pokemon->cantidad_pokemones) - 1] =
+			calloc(1, sizeof(pokemon_t));
+		*(info_pokemon
+			  ->pokemones[(info_pokemon->cantidad_pokemones) - 1]) =
+			*nuevo_pokemon;
+	} else {
+		*seguir_leyendo = false;
+		printf("ocurrio una falla al usar realloc\n");
+	}
+}
+
 informacion_pokemon_t *pokemon_cargar_archivo(const char *path)
 {
-	return NULL;
+	if (path == NULL) {
+		return NULL;
+	}
+
+	FILE *archivo = fopen(path, "r");
+
+	if (archivo == NULL) {
+		return NULL;
+	}
+
+	informacion_pokemon_t *info_pokemon =
+		calloc(1, sizeof(informacion_pokemon_t));
+	if (info_pokemon == NULL) {
+		printf("error al asignar memoria con calloc para info_pokemon");
+		fclose(archivo);
+		return NULL;
+	}
+
+	bool seguir_leyendo = true;
+
+	while (seguir_leyendo) {
+		char linea_pokemon[MAX_CARACTERES];
+		char lineas_ataques[MAX_ATAQUES][MAX_CARACTERES];
+		if (leer_lineas(archivo, linea_pokemon, lineas_ataques) == 4) {
+			pokemon_t *nuevo_pokemon = crear_nuevo_pokemon(
+				linea_pokemon, lineas_ataques);
+			if (nuevo_pokemon != NULL) {
+				asignar_pokemon_a_info_pokemon(info_pokemon,
+							       nuevo_pokemon,
+							       &seguir_leyendo);
+				free(nuevo_pokemon);
+			} else {
+				seguir_leyendo = false;
+			}
+		} else {
+			seguir_leyendo = false;
+		}
+	}
+
+	fclose(archivo);
+
+	if (info_pokemon->cantidad_pokemones >= 1) {
+		return info_pokemon;
+	} else {
+		pokemon_destruir_todo(info_pokemon);
+		return NULL;
+	}
+}
+
+void recorrer_y_buscar_nombre(informacion_pokemon_t *ip, const char *nombre,
+			      int *indice_pokemon_encontrado)
+{
+	int i = 0;
+	while (*indice_pokemon_encontrado == -1 && i < ip->cantidad_pokemones) {
+		if (strcmp(ip->pokemones[i]->nombre, nombre) == 0) {
+			*indice_pokemon_encontrado = i;
+		}
+		i++;
+	}
 }
 
 pokemon_t *pokemon_buscar(informacion_pokemon_t *ip, const char *nombre)
 {
-	return NULL;
+	if (ip == NULL || nombre == NULL) {
+		return NULL;
+	}
+
+	int indice_pokemon_encontrado = -1;
+	recorrer_y_buscar_nombre(ip, nombre, &indice_pokemon_encontrado);
+
+	return (indice_pokemon_encontrado != -1) ?
+		       ip->pokemones[indice_pokemon_encontrado] :
+		       NULL;
 }
 
 int pokemon_cantidad(informacion_pokemon_t *ip)
 {
-	return 0;
+	if (ip == NULL) {
+		return 0;
+	}
+
+	return ip->cantidad_pokemones;
 }
 
 const char *pokemon_nombre(pokemon_t *pokemon)
 {
-	return NULL;
+	if (pokemon == NULL) {
+		return NULL;
+	}
+
+	return pokemon->nombre;
 }
 
 enum TIPO pokemon_tipo(pokemon_t *pokemon)
 {
-	return FUEGO;
+	if (pokemon == NULL) {
+		return NORMAL;
+	}
+
+	return pokemon->tipo_pokemon;
+}
+
+void recorrer_y_buscar_ataque(pokemon_t *pokemon, const char *nombre,
```

Comment:
lo mismo, este codigo podria esta en pokemon_buscar_ataque y usar un for en vez de un while.